src/core/ddsc/src/dds_qos.c

```diff
@@ -565,3 +710,91 @@ bool dds_qget_ignorelocal (const dds_qos_t * __restrict qos, dds_ignorelocal_kin
     *ignore = qos->ignorelocal.value;
   return true;
 }
+
+bool dds_qget_propnames (const dds_qos_t * __restrict qos, uint32_t * n, char *** names)
+{
+  bool props;
+  if (qos == NULL || (n == NULL && names == NULL))
+    return false;
+
+  props = (qos->present & QP_PROPERTY_LIST) && qos->property.value.n > 0;
+
+  if (n != NULL)
+    *n = props ? qos->property.value.n : 0;
+
+  if (names != NULL)
+  {
+    if (!props || qos->property.value.n == 0)
+      *names = NULL;
+    else
+    {
+      *names = dds_alloc (sizeof (char *) * qos->property.value.n);
+      for (uint32_t i = 0; i < qos->property.value.n; i++)
+        (*names)[i] = dds_string_dup (qos->property.value.props[i].name);
+    }
+  }
+  return props;
+}
+
+bool dds_qget_prop (const dds_qos_t * __restrict qos, const char * name, char ** value)
+{
+  uint32_t i;
+  bool found;
+
+  if (qos == NULL || name == NULL)
+    return false;
+
+  found = dds_qprop_get_index (qos, name, &i);
+  if (value != NULL)
+    *value = found ? dds_string_dup (qos->property.value.props[i].value) : NULL;
+  return found;
+}
+
+bool dds_qget_bpropnames (const dds_qos_t * __restrict qos, uint32_t * n, char *** names)
```

Comment:
Much like ... etc. — you can guess the rest 🙂

Reply:
Fixed, see above